src/NBitcoin/Block.cs

```diff
@@ -357,7 +387,7 @@ public bool Check()
         /// <returns></returns>
         public bool Check(Consensus consensus)
         {
-            if (Block.BlockSignature)
+            if (this.GetNetworkOptions().IsProofOfStake)
```

Comment:
If Options was in consensus we would be in a good place maybe consider this?

Reply:
I'm slightly biased towards a consistent approach of having the options flow from network -> block -> transaction with interception points along the way. I have to try hard to convince myself of the benefits of having it on the consensus. However nothing prevents us from looking at both network and consensus when compiling the NetworkOptions that will be associated with the Network object.